### ISSUE TITLE
Issues with input parser with items that begin with a number

### ISSUE THREAD

**Issue by dome2048** [openhab/openhab-webui#586]
## The problem
The UI is throwing up errors when items begin with a number, i.e. a group name "1F" or an item named "1F_Scene"

## Expected behavior
That the UI wouldn't throw up errors for improper item names when they begin with a number. The sitemap and items perform properly when created other ways so this appears to be only an issue with the UI and not the back end.

## Steps to reproduce

I'm experiencing this in two primary places so far: creating items textually and creating/editing a sitemap in the MainUI.

**Item Creation:**
I began creating items by using the "create item from textual definition" and copy/pasting them from my existing OH2 items files. Items that began with a number threw up errors. Here is an example:

`Switch 3SUN_HolidayLights`

Error:

              Error: Syntax error at line 1 col 8:

                Switch 3
                       ^
              Unexpected number token: "3". Instead, I was expecting to see one of the following:

note- log truncated

Changing the item name to not begin with a number does eliminate the error but violates the naming convention I'd like to use. When created a different way, such as using the simple "create item" entry or by selecting "create points/equipment from thing" under the model, this does not seem to cause a problem and I've successfully created items that begin with numbers this way.

**Sitemap editing:**
The offending line of code:

`Selection item=1F_Scene mappings=[100="All On",0="All Off",12="Ambient",22="Bright",25="Workshop"]`

Error log:

              Error: Syntax error at line 85 col 29:
              Selection item=1F_Scene
                              ^
              Unexpected identifier token: "F_Scene". Instead, I was expecting to see one of the following:

note- remainder of logging output truncated as it appeared purely instructional.

This sitemap works properly when loaded, and in fact the error didn't appear until I changed the item name in the GUI editor, then switched over to the code view.

## Your environment
openHAB 3 M3, run in Docker container on Ubuntu 20.04.1 LTS

> runtimeInfo:
>   version: 3.0.0.M3
>   buildString: Milestone Build
> locale: en_US
> systemInfo:
>   configFolder: /openhab/conf
>   userdataFolder: /openhab/userdata
>   logFolder: /openhab/userdata/logs
>   javaVersion: 11.0.8
>   javaVendor: Azul Systems, Inc.
>   javaVendorVersion: Zulu11.41+23-CA
>   osName: Linux
>   osVersion: 5.4.0-54-generic
>   osArchitecture: amd64
>   availableProcessors: 12
>   freeMemory: 457803520
>   totalMemory: 761266176
> bindings:
>   - astro
>   - hue
>   - mqtt
>   - remoteopenhab
> clientInfo:
>   device:
>     ios: false
>     android: false
>     androidChrome: false
>     desktop: true
>     iphone: false
>     ipod: false
>     ipad: false
>     edge: false
>     ie: false
>     firefox: true
>     macos: true
>     windows: false
>     cordova: false
>     phonegap: false
>     electron: false
>     nwjs: false
>     webView: false
>     webview: false
>     standalone: false
>     os: macos
>     pixelRatio: 0.8955223880597015
>     prefersColorScheme: dark
>   isSecureContext: false
>   locationbarVisible: true
>   menubarVisible: true
>   navigator:
>     cookieEnabled: true
>     deviceMemory: N/A
>     hardwareConcurrency: 2
>     language: en-US
>     languages:
>       - en-US
>       - en
>     onLine: true
>     platform: MacIntel
>   screen:
>     width: 2144
>     height: 1206
>     colorDepth: 24
>   support:
>     touch: false
>     pointerEvents: true
>     observer: true
>     passiveListener: true
>     gestures: false
>     intersectionObserver: true
>   themeOptions:
>     dark: dark
>     filled: true
>     pageTransitionAnimation: default
>     bars: light
>     homeNavbar: simple
>     homeBackground: default
>     expandableCardAnimation: default
>   userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:80.0) Gecko/20100101
>     Firefox/80.0
> timestamp: 2020-12-02T07:15:27.926Z
> 

```yaml

```

## Browser console

```txt

```

**Comment by dome2048** [comment]
Just confirmed that this bug is still present in Snapshot #2037.

**Comment by hmerk** [comment]
This is not clearly not a bug. If you read the docs carefully, you might see that item names must not start with a number...

The only characters permitted in an Item name are letters, numbers and the underscore character. Names must not begin with numbers.

**Comment by dome2048** [comment]
After opening the issue, I did indeed find the documentation page where it states that item names are not allowed to start with a number. Perhaps I used to know this, but as a longtime user it's one of those facts I didn't remember. 

IMHO, this would be incredibly frustrating to a brand new openHAB user to be allowed to create a model that is broken from the start. I doubt a brand new user would understand when they were creating locations they were even creating items, much less think to read the docs page to ensure their naming conventions lined up. Especially since I'm sure many other people will come along and try to create items like "1F" or "1st_Floor". Even more confusing is that openHAB actually did seem to run properly with all these items incorrectly defined, and the only place I saw errors was in the above mentioned GUI editors. 

I propose therefore a better check when creating locations, equipment, or points that will disallow creation of an item stat starts with a number.
